opentelemetry-api/src/opentelemetry/attributes/__init__.py

```diff
@@ -172,14 +172,16 @@ def __setitem__(self, key, value):
                 self.dropped += 1
                 return
 
-            if key in self._dict:
-                del self._dict[key]
-            elif self.maxlen is not None and len(self._dict) == self.maxlen:
-                del self._dict[next(iter(self._dict.keys()))]
-                self.dropped += 1
-
             value = _clean_attribute(key, value, self.max_value_len)
             if value is not None:
+                if key in self._dict:
+                    del self._dict[key]
+                elif (
+                    self.maxlen is not None and len(self._dict) == self.maxlen
+                ):
+                    self._dict.popitem(last=False)
```

Comment:
What does `last=False` do? Convert it to FIFO?

Reply:
Yes